static/js/drafts.js

```diff
@@ -431,8 +496,76 @@ export function launch() {
         });
     }
 
-    const drafts = format_drafts(draft_model.get());
-    render_widgets(drafts);
+    function filter_drafts_by_compose_box_and_recipient(drafts) {
+        // Prioritize stream, topic, and PM recipients from the compose box first and then
+        // from the narrow.
+        const composing = compose_state.composing();
+        let stream = null;
+        let topic = null;
+        if (composing) {
+            if (compose_state.get_message_type() === "stream") {
+                stream = compose_state.stream_name();
+                topic = compose_state.topic();
+            }
+        } else {
+            stream = narrow_state.stream();
+            topic = narrow_state.topic();
+        }
+
+        let private_recipients = null;
+        if (composing) {
+            if (compose_state.get_message_type() === "private") {
+                private_recipients = compose_state.private_message_recipient();
+            }
+        } else {
+            private_recipients = narrow_state.pm_email_string();
+        }
+
+        const narrow_drafts_ids = [];
+        for (const [id, draft] of Object.entries(drafts)) {
+            // Match by stream and topic.
+            if (
+                draft.type === "stream" &&
+                stream &&
+                topic &&
+                draft.stream === stream &&
+                draft.topic === topic
```

Comment:
This algorithm is wrong, since it's an incorrectly case-sensitive comparison.

Would `util.same_recipient` be helpful here?

Reply:
Oh shoot I meant to use that here when you mentioned it in the other PR -- thanks for bringing it up again!